source/assets/components/RecipeCard.js

```diff
@@ -0,0 +1,273 @@
+  }
+
+  article time {
+    color: #305A50;
+    font-size: 18px;
+    text-align: center;
+    position:relative;
+    bottom: 20px;
+  }
+
+  article:hover .cook{
+    background: darkgreen;
+    color: white;
+  `;
+  styleElem.innerHTML = styles;
+
+  //recipe card
+  const card = document.createElement("article");
+  
+  //recipe card image
+  const recipeImg = document.createElement("img");
+  recipeImg.src = searchForKey(data, "thumbnail") || searchForKey(data, "thumbnailUrl") || searchForKey(data, "url");
+  recipeImg.alt = searchForKey(data, "headline");
+  card.appendChild(recipeImg);
+
+  //recipe card favorite button
+  const input = document.createElement("input");
+    let liked = true;
+    input.classList.add("favorite");
+    input.type = "image";
+    input.setAttribute("src", "assets/images/heart.png");
+    
+    input.addEventListener("click", changeHeart);
+    
+    /**
+     * Toggle on and off the favorite heart button
+     * @returns void
+     */
+    function changeHeart()
+    {
+      if (liked)
+      {
+          input.setAttribute("src", "assets/images/empty_heart.png");
+          liked = false;
+      }
+      else
+      {
+          input.setAttribute("src", "assets/images/heart.png");
+          liked = true;
+      }
+    }
+
+    card.appendChild(input);
+
+  // p element - title
+  const recipeTitle = document.createElement("p");
+  recipeTitle.classList.add("title");
+  recipeTitle.textContent = searchForKey(data, "headline");
+  card.appendChild(recipeTitle);
+
+  //time element
+  const time = document.createElement("time");
+  const totalTime = searchForKey(data, "totalTime");
+  time.textContent = convertTime(totalTime);
+  card.appendChild(time);
+
+  // button element - check recipe
+  const checkButton = document.createElement("button");
+  checkButton.classList.add("cook");
+  checkButton.textContent = "View Recipe  ";
+  card.appendChild(checkButton);
+  checkButton.addEventListener("click", function (event) {
+    showCheck(data);
+  });
+
+   this.shadowRoot.appendChild(styleElem);
+   this.shadowRoot.appendChild(card);
+
+  }
+}
+
+
+/**
+ * Recursively search for a key nested somewhere inside an object
+ * @param {Object} object the object with which you'd like to search
+ * @param {String} key the key that you are looking for in the object
+ * @returns {*} the value of the found key
+ */
+function searchForKey(object, key) {
+  var value;
+  Object.keys(object).some(function (k) {
+    if (k === key) {
+      value = object[k];
+      return true;
+    }
+    if (object[k] && typeof object[k] === 'object') {
+      value = searchForKey(object[k], key);
+      return value !== undefined;
+    }
+  });
+  return value;
+}
+
+/**
+ * Extract the URL from the given recipe schema JSON object
+ * @param {Object} data Raw recipe JSON to find the URL of
+ * @returns {String} If found, it returns the URL as a string, otherwise null
+ */
+function getUrl(data) {
+  if (data.url) return data.url;
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/cse110-fa21-group16/cse110-fa21-group16/pullRequest?prid=8394473)
```suggestion
  if (data.url) {return data.url;}
```